boot/bootutil/src/bootutil_misc.c

```diff
@@ -170,14 +171,20 @@ boot_magic_off(const struct flash_area *fap)
 }
 
 int
-boot_status_entries(const struct flash_area *fap)
+boot_status_entries(int image_index, const struct flash_area *fap)
 {
+#if (BOOT_IMAGE_NUMBER == 1)
+    (void)image_index;
```

Comment:
Would it be useful to put the `(void)image_index` into `FLASH_AREA_IMAGE_PRIMARY` and `SECONDARY` when there is only a single image. After all, it is the macro that isn't using its argument. This would avoid having to add this block to a bunch of functions.

Reply:
Although, this could fairly easily be fixed in another commit. It might be nice to figure if there are other of these "unused" blocks that can be eliminated this way, and to figure the best way to "use" a macro argument.

Reply:
Nice one, I changed the definition to:

```
#define FLASH_AREA_IMAGE_PRIMARY(x)    (((x) == 0) ?                \           
                                         DT_FLASH_AREA_IMAGE_0_ID : \           
                                         DT_FLASH_AREA_IMAGE_0_ID)
```
And removed all `(void)image_index`...

Reply:
As long as we don't build with too stupid of a compiler, this should be fine.